src/main/java/org/gridsuite/loadflow/server/utils/SpecificationBuilder.java

```diff
@@ -32,12 +37,27 @@ private static <T> Specification<T> buildSpecifications(
             UUID uuid,
             List<ResourceFilter> filters,
             String fieldName,
-            String uuidFieldName
+            String uuidFieldName,
+            boolean withJoin
     ) {
         return (root, query, criteriaBuilder) -> {
             List<Predicate> predicates = new ArrayList<>();
-            filters.forEach(filter -> SpecificationUtils.addPredicate(criteriaBuilder, root, predicates, filter));
+            // user filters on main entity
+            filters.stream().filter(SpecificationBuilder::isParentFilter).forEach(filter -> SpecificationUtils.addPredicate(criteriaBuilder, root, predicates, filter));
             predicates.add(root.get(fieldName).get(uuidFieldName).in(List.of(uuid)));
+            List<ResourceFilter> childrenFilters = filters.stream().filter(resourceFilter -> !isParentFilter(resourceFilter)).toList();
+            if (withJoin) {
```

Comment:
I see that withJoin is used for LoadFlowResult. Why not create a separate buildSpecification, one for LimitViolation and one for LoadFlowResult, and factorize the common code?

Reply:
WithJoin is used for ComponentResultEntity not for LoadFlowResult. 
buildSpecification is already used for both buildLimitViolationsSpecifications and buildLoadflowResultSpecifications  so i'm not sure that i get you point.